### PR TITLE
Use dynamic length array in FFI to fix the `make test` error

### LINKED ISSUE #17: `make test` error
I run `make test`, and get error info such as:
```
C wrapper, verifier parameter deserialization failed:Custom { kind: InvalidData, error: NotInSubgroup }
```
OR
```
C wrapper, verifier parameter deserialization failed:Custom { kind: InvalidData, error: NotOnCurve }
```
While run `cargo test test_c_api_basic` is okay, seemed something wrong in the FFI function due to my environment?
My gcc and rustc version info:
rustc 1.37.0 (eae3437df 2019-08-13)
gcc version 5.4.0 20160609 (Ubuntu 5.4.0-6ubuntu1~16.04.12)

### PR DESCRIPTION
**Pull request by 3for**
Use dynamic length array in FFI to fix #17 .

### CODE REVIEW

**Comment by zhenfeizhang** [comment]
Is #17 still an issue for you after 102df1b898a5565d4bc93854819527b18d8fa5cb?
It appears fine on my end.

**Comment by 3for** [comment]
#17 is okay after [102df1b](https://github.com/algorand/pointproofs/commit/102df1b898a5565d4bc93854819527b18d8fa5cb), but still an issue while clear the `GROUP_SWITCHED` setting in `cbindgen.toml`.
It's said that operations in G1 are typically faster than in G2.
I've debugged the code with `GROUP_SWITHCED` undefined, the array data in Rust is different while transferred to C since the 49738-th index (start from 1)。
And it would be more  flexible to use dynamic array in FFI for any vector length instead of fixing to `n=1024`.

**Comment by zhenfeizhang** [comment]
I think it is better to use "GROUP_SWITCHED" macros than dynamic arrays: we want the caller to know if they want the group to be switched, and enforcing the C caller to defined it is one extra sanity check that the caller knows about if group is switched or not.

Agree with the flexibility limitation for `n=1024`.

**Comment by 3for** [comment]
Agree with the use of  "GROUP_SWITCHED" macros, and it's not conflict with dynamic arrays.
Dynamic array is used to fix the flexibility limitation, so `n` can be any value instead of `1024`. Setting `n=16` in `test_dynamic.c` will also let the `make dtest` pass.
Seemed something wrong happens in the FFI when the fixed-length array is longer than 49738.

**Comment by zhenfeizhang** [comment]
Okay. I guess it makes sense to make the change for future proof. Please
* remove old `c_api` and rename `c_api_dynamic` to `c_api` (also for the tests, structures, etc.)
* make sure that the code passes our travis test

File: src/Cargo.toml [hunk 1]
```diff
@@ -0,0 +1,59 @@
+[package]
```
Comment (zhenfeizhang) [inline src/Cargo.toml:1]: This file, as well as a few other files under 'src' seems to be duplicates

Comment (3for) [reply src/Cargo.toml]: Sorry for the uncareful git commit therefore.

File: c_test/pointproofs_c.h [hunk 2]
```diff
@@ -8,10 +8,16 @@
 #include <stdlib.h>
 
 #if !defined(GROUP_SWITCHED)
+/**
```
Comment (zhenfeizhang) [inline c_test/pointproofs_c.h:11]: `pointproofs_c.h` is auto-generated. Those comments are not generated from `cbindgen`

Comment (3for) [reply c_test/pointproofs_c.h:11]: I've tried again, it's actually auto-generated on my server. 
`cbindgen -V`: 
```
cbingen 0.14.2
```

**Pull request merged**